jobs/src/main/scala/dcos/metronome/jobrun/impl/JobRunServiceActor.scala

```
@@ -42,31 +41,40 @@ class JobRunServiceActor(
 
     // trigger messages
-    case TriggerJobRun(spec, startingDeadline) => triggerJobRun(spec, startingDeadline)
+    case TriggerJobRun(spec, schedule, startingDeadline) => triggerJobRun(spec, schedule, startingDeadline)
 
     // executor messages
-    case JobRunUpdate(started)                 => updateJobRun(started)
-    case Finished(result)                      => jobRunFinished(result)
-    case Aborted(result)                       => jobRunFailed(result)
-    case Failed(result)                        => jobRunFailed(result)
+    case JobRunUpdate(started)                           => updateJobRun(started)
+    case Finished(result)                                => jobRunFinished(result)
+    case Aborted(result)                                 => jobRunFailed(result)
+    case Failed(result)                                  => jobRunFailed(result)
 
     //event stream events
-    case update: TaskStateChangedEvent         => forwardStatusUpdate(update)
+    case update: TaskStateChangedEvent                   => forwardStatusUpdate(update)
   }
 
   def runsForJob(jobId: JobId): Iterable[StartedJobRun] = allJobRuns.values.filter(_.jobRun.id.jobId == jobId)
 
-  def triggerJobRun(spec: JobSpec, startingDeadline: Option[Duration]): Unit = {
+  def triggerJobRun(spec: JobSpec, schedule: Option[ScheduleSpec], startingDeadline: Option[Duration]): Unit = {
     log.info(s"Trigger new JobRun for JobSpec: $spec")
-    val jobRun = JobRun(JobRunId(spec), spec, JobRunStatus.Initial, clock.now(), None, startingDeadline, Map.empty)
-    val startedJobRun = startJobRun(jobRun)
-    sender() ! startedJobRun
+
+    val skipRun = schedule match {
+      case Some(schedule) => schedule.concurrencyPolicy == ConcurrencyPolicy.Forbid && runsForJob(spec.id).nonEmpty
+      case None           => false
+    }
+    if (skipRun) {
+      log.debug(s"SkippingB Scheduled run for ${spec.id} based on concurrency policy")
```

silently dropped is fine?

---

I would log this as info

Also this is a typo SkippingB
And I would start the word Scheduled with lovercase letter

---

I much prefer adding to the logs what this event happened... it is a debug message and not an info